flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3208,12 +3248,15 @@ def test_listed_volume_attributes(self):
             dataset_id=expected_dataset_id,
             size=self.minimum_allocatable_size,
         )
-        [listed_volume] = self.api.list_volumes()
 
-        self.assertEqual(
-            (expected_dataset_id, self.minimum_allocatable_size),
-            (listed_volume.dataset_id, listed_volume.size)
-        )
+        def validate():
+            [listed_volume] = self.api.list_volumes()
```

Comment:
This sometimes fails because `list_volumes` returns an empty list due to eventual consistency. Please make it so we retry in that case too.